packages/ui/src/widgets/Preloader/Preloader.tsx

```diff
@@ -57,13 +58,21 @@ const StartButton = styled(Button)(
 export type PreloaderProps = {
   ready?: boolean;
   onStartClick?: () => Promise<void> | void;
+  navigateToNextScreen?: () => void;
 };
 
-export const Preloader = ({ ready = false, onStartClick }: PreloaderProps) => {
+export const Preloader = ({ ready = false, onStartClick, navigateToNextScreen }: PreloaderProps) => {
   const isLandscape = useMediaQuery('(max-height: 440px)');
   const [handleStartClick, isBusy] = useAsyncCallback(onStartClick);
   const { loading, preloader } = useGameInfo();
   const { sdkUrl: cdnUrl } = useConfigContext();
+  const { isReady, address } = useWalletContext();
+
+  const walletStatus = useMemo(() => (isReady ? 'READY' : 'NOT_READY'), [isReady]);
+
+  if (walletStatus === 'READY' && address) {
```

Comment:
To achieve the same results it will be enough to just check `address`, so you don't actually need `isReady` and `walletStatus`

```suggestion
  if (address) {
```